azkaban-common/src/test/java/azkaban/executor/MockExecutorLoader.java

```diff
@@ -122,6 +122,9 @@ public void uploadExecutableNode(ExecutableNode node, Props inputParams)
   @Override
   public void updateExecutableNode(ExecutableNode node)
       throws ExecutorManagerException {
+    if (!nodes.containsKey(node.getId())) {
+      uploadExecutableNode(node, new Props());
+    }
```

Comment:
After updating latest changes from `master`, the test was failing here because the node wasn't found in the map. That could be just concurrency issue, because here `nodes` is just a regular `HashMap` without thread safety.. Or then there's an actual change in behaviour that makes an update before initial upload. I wonder if that would be a problem when running against the actual DB layer..